db/migrations/20230227122704-ss58-prefix.js

```diff
@@ -0,0 +1,222 @@
+
+            if (identity && identity.includes("parent")) {
+                const currentParent = JSON.parse(identity).parent;
+
+                const nextParent = keyring.encodeAddress(
+                    keyring.decodeAddress(currentParent),
+                    ss58Format);
+
+                const nextIdentity = identity.replace(currentParent, nextParent);
+
+                queryString += `, identity='${nextIdentity}'`
+            }
+
+            queryString += ` WHERE rank='${rank}'`;
+
+            db.runSql(queryString, logger.log);
+        });
+
+    });
+}
+
+const convertFaucetTable = (db, ss58Format) => {
+    const logger = new Logger('faucet');
+
+    db.runSql('SELECT id, sender, destination from faucet;', (_, result) => {
+        logger.setTotalRows(result.rows.length);
+
+        result.rows.forEach(({id, sender, destination}) => {
+            const nextSender = keyring.encodeAddress(
+                keyring.decodeAddress(sender),
+                ss58Format);
+
+            const nextDestination = keyring.encodeAddress(
+                keyring.decodeAddress(destination),
+                ss58Format);
+
+            const queryString = `UPDATE faucet SET sender='${nextSender}', destination='${nextDestination}' WHERE id='${id}';`
+
+            db.runSql(queryString, logger.log);
+        });
+
+    });
+}
+
+// Transfers are a part of extrinsict table
+const convertTransfers = (db, ss58Format) => {
+    const logger = new Logger('transfer');
+
+    db.runSql(`select block_number, signer, args, method from extrinsic where method like 'transfer%'`, (_, result) => {
+        logger.setTotalRows(result.rows.length);
+
+        result.rows.forEach(({block_number, signer, args, method}) => {
+            const nextSigner = keyring.encodeAddress(
+                keyring.decodeAddress(signer),
+                ss58Format);
+
+            let queryString = `UPDATE extrinsic SET signer='${nextSigner}'`
+
+            if (method === 'transfer' || method === 'transferKeepAlive') {
+                const [account] = JSON.parse(args);
+                const accountId = typeof account === 'string' ? account : account.id;
+                const nextAccountId = keyring.encodeAddress(
+                    keyring.decodeAddress(accountId),
+                    ss58Format);
+                const nextArgs = args.replace(accountId, nextAccountId);
+                queryString += `, args='${nextArgs}'`;
+            }
+
+            queryString += ` WHERE block_number='${block_number}';`
+
+            db.runSql(queryString, logger.log);
+        });
+
+    });
+};
+
+const convertTables = (db, prefix) => {
+    convertAccountTable(db, prefix);
+    convertEraTables(db, prefix);
+    convertRankingTable(db, prefix);
+    convertFaucetTable(db, prefix);
+    convertTransfers(db, prefix);
+}
+
+exports.up = async function (db) {
+    const newSs58Prefix = 54;
+    convertTables(db, newSs58Prefix);
```

Comment:
What behaviour will be if to execute this method with address with new prefix?

Reply:
I checked it. Actually, nothing because if we try to convert the address with prefix X to prefix X it will be the same result.